cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -204,48 +205,49 @@ public boolean rollbackTx() throws Exception {
   }
 
   private void rollbackOperation(PartitionOperation operation) throws Exception {
-    switch (operation.getOperationType()) {
-      case CREATE:
-        undoPartitionCreate(operation.getPartitionKey(), operation.getRelativePath());
-        break;
-      case DROP:
-        undoPartitionDelete(operation.getPartitionKey(), operation.getRelativePath());
-        break;
-      default:
-        // this shouldn't happen, since we only have the above types defined in the Enum.
-        throw new IllegalArgumentException("Unknown operation type: " + operation.getOperationType());
+    if (operation instanceof AddPartitionOperation) {
+      undoPartitionCreate((AddPartitionOperation) operation);
+    } else if (operation instanceof DropPartitionOperation) {
+      undoPartitionDelete((DropPartitionOperation) operation);
+    } else {
+      // this shouldn't happen, since we only have the above types defined in the Enum.
```

Comment:
nit - comment is outdated; there's no longer any such Enum, so it can be confusing